balloon-hash/src/balloon.rs

```diff
@@ -179,25 +185,30 @@ where
                     Digest::update(&mut digest, thread_id.to_le_bytes());
                 }
 
-                Digest::update(&mut digest, idx_block);
+                Digest::update(&mut digest, &idx_block);
+                idx_block.zeroize();
```

Comment:
Does this make sense?

Reply:
After some discussion on Zulip I decided to only zeroize things that are really necessary. Like the hash output.